.travis.yml

```
@@ -0,0 +1,29 @@
+---
+language: python
+python: "2.7"
+
+# Use the new container infrastructure
+sudo: false
+
+# Install ansible
+addons:
+  apt:
+    packages:
+    - python-pip
+
+install:
+  # Install ansible
+  - pip install ansible
+
```

Should we add the requirements for managing windows such as pip to install the requests-credssp library? I'm sure we will tackle this as we move towards running playbooks against windows instances. Just a thought.
```suggestion
pip install pywinrm
pip install pywinrm[credssp]
```

---

I'm not sure this would be useful in the CI/CD Pipeline unless we were deploying servers to test against during the testing. Is that the later intent or are you just thinking that it would be nice to already have baked into the template?

---

The thought is that we will eventually be utilizing this ansible method against windows hosts and for ansible to communicate to windows over winrm the pywinrm python module needs to be installed on the ansible control node.

---

I've been thinking about how we would do testing w/ Windows. Server Core version of Windows Server 2019 Datacenter Edition is available for testing against Windows in CircleCI, but creating a Windows container inside a Linux container isn't supported, which is what we would need to do to run Ansible (either directly or with Molecule) against Windows, since Windows can't run ansible on localhost.

---

Yeah the inability to run ansible on windows is a huge gap. That makes everything more complicated than it should be. I'm not sure the best move here but we can always add it to the baseline at a later date.

---

Works for me sounds like an interesting challenge.

---

Also, the testing in the `.travis.yml` file is just doing a syntax check, and not actually running the playbook, so it doesn't matter what OS it is on. Actual testing of the playbooks (say, with Molecule) isn't part of this template (yet)